torchbox/torchbox.com
# WIP: FE | Impact Report page styles

[Link to Ticket](https://torchbox.monday.com/boards/1192293412/pulses/1439824403)
Design: https://www.figma.com/proto/QdMXBHr57mN1svsaNQ7Ggj/Torchbox-2024-(Master)?type=design&node-id=5786-37133&viewport=-9122%2C427%2C0.25&t=TL743gKILilhCSZb-0&scaling=scale-down-width&starting-point-node-id=5786%3A37133&show-proto-sidebar=1&hide-ui=1 (there was no mobile design)

### Description of Changes Made

This PR aims to style the impact report page in the new site design. The idea is to use a lot of the existing blocks in the site but add a couple that Olly particularly wanted - the instagram and small image with text block.

We discussed making the in-page navigation menu non-sticky for speed but it didn't work very well like that. I have implemented a very basic sticky menu with CSS only - may get enhanced with javascript (more like the old one) down the line.

There are a few back-end changes in order to bring it in line with the design.

Authored by both @SharmaineLim and @helenb 

- [x] Set up dummy blocks for the impact report page. (This allows us to style while looking at [the page in the pattern library](http://localhost:8000/pattern-library/render-pattern/patterns/pages/impact_reports/impact_report_page.html).)
- [x] Align existing standard blocks to the impact report page's "center" / content area.
- [x] Style the impact report heading block.
- [x] Style the small image with text block.
- [x] Style the instagram block
- [x] Styling for the heading area
- [x] Styling for main image
- [x] Anchor links

Other changes:

- Add `get_verbose_name` to homepage.yaml file so that homepage-specific styles apply on the pattern library.
- Rename instagram gallery block file for consistency.
- Removes lazy loading from the work page main image as it partly above the fold.

### How to Test

Create a new impact report page in your local build, and populate it similarly to the design.

Below are all the relevant pattern library links, which will work, but won't have the correct style of imagery for heading images and the small image with text icons.

http://localhost:8000/pattern-library/pattern/patterns/molecules/streamfield/blocks/instagram_gallery_block.html
http://localhost:8000/pattern-library/pattern/patterns/molecules/streamfield/blocks/small_image_with_text_block.html
http://localhost:8000/pattern-library/pattern/patterns/molecules/streamfield/blocks/impact_report_heading_block.html
http://localhost:8000/pattern-library/pattern/patterns/pages/impact_reports/impact_report_page.html
http://localhost:8000/pattern-library/render-pattern/patterns/styleguide/components/components.html

Note that ideally for the 'small image with text block', ideally the editor would be given a choice of icons which would automatically update their colours based on the theme. But due to time constraints we have gone with a simple png upload instead - the theme of the page is unlikely to change (not the design shows it in 'banana' theme). Icons can be found here: https://drive.google.com/drive/folders/1_Q1dPJBw5GyAeUFhLddtG0EZvf4RyjyF

If you pull the production images you can find examples of the heading images in the 'Impact report' collection (same ones as were used on the previous site).

### Screenshots

<details><summary>Impact report heading block</summary>

![image](https://github.com/torchbox/torchbox.com/assets/5073278/dd4b5d35-2a00-47c5-88a8-ca959ea7a4bb)

![image](https://github.com/torchbox/torchbox.com/assets/5073278/6a0bda44-d14e-47d6-8e37-4790ebd6f0af)

</details>

<details><summary>Small image with text block</summary>

<img width="838" alt="Screenshot 2024-04-11 at 13 58 17" src="https://github.com/torchbox/torchbox.com/assets/771869/0b952449-47e5-43f2-bf54-5375d03b2463">


</details>

<details><summary>Impact report sticky navigation</summary>

<img width="1436" alt="Screenshot 2024-04-11 at 13 48 54" src="https://github.com/torchbox/torchbox.com/assets/771869/231c6a6b-2b86-451c-94d4-8ba412050f64">

<img width="344" alt="Screenshot 2024-04-11 at 14 03 29" src="https://github.com/torchbox/torchbox.com/assets/771869/bf5dfdb7-2a55-4de5-bb0f-b14e4d092bfc">


</details>

<details><summary>Instagram block</summary>

<img width="597" alt="Screenshot 2024-04-11 at 13 56 17" src="https://github.com/torchbox/torchbox.com/assets/771869/b49b475c-3579-4217-9cce-45d3cb43e33b">


</details>

### MR Checklist


#### Unit tests


#### Documentation


#### Browser testing

  - Latest version of Chrome on mac
  - Latest version of Firefox on mac
  - Latest version of Safari on mac
  - Safari on last two versions of iOS
  - Chrome on last two versions of Android

#### Data protection


#### Accessibility


#### Sustainability


#### Pattern library
